### PR TITLE
[13.0][IMP] web_widget_one2many_product_picker: Refactor code

### PR DESCRIPTION
**Pull request by Tardo**
- All calls to onchange are now 'shadowed'. So, no more "loading overlay" is showed.
- Now the elements are displayed in the correct order.
- Now the record is blocked while the request is being processed on the server
- Avoid 'sale.order' onchanges for 'order_line' subfields to improve the performance
- More friendly with users showing status messages
- Consolidated autosave functionality

cc @tecnativa TT32568

### CODE REVIEW

**Comment by pedrobaeza** [comment]
@Tardo please squash and fix pre-commit.

**Comment by Tardo** [comment]
@pedrobaeza done and linters green.

File: web_widget_one2many_product_picker/README.rst [hunk 1]
```diff
@@ -14,13 +14,13 @@ Web Widget One2Many Product Picker
     :target: http://www.gnu.org/licenses/agpl-3.0-standalone.html
     :alt: License: AGPL-3
 .. |badge3| image:: https://img.shields.io/badge/github-OCA%2Fweb-lightgray.png?logo=github
-    :target: https://github.com/OCA/web/tree/13.0/web_widget_one2many_product_picker
+    :target: https://github.com/OCA/web/tree/12.0/web_widget_one2many_product_picker
     :alt: OCA/web
 .. |badge4| image:: https://img.shields.io/badge/weblate-Translate%20me-F47D42.png
-    :target: https://translation.odoo-community.org/projects/web-13-0/web-13-0-web_widget_one2many_product_picker
+    :target: https://translation.odoo-community.org/projects/web-12-0/web-12-0-web_widget_one2many_product_picker
```
Comment (pedrobaeza) [inline web_widget_one2many_product_picker/README.rst]: This shouldn't be changed

Comment (Tardo) [reply web_widget_one2many_product_picker/README.rst]: Changed

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -15,7 +15,7 @@ exclude: |
   # You don't usually want a bot to modify your legal texts
   (LICENSE.*|COPYING.*)
 default_language_version:
-  python: python3
+  python: python3.9
```
Comment (pedrobaeza) [inline .pre-commit-config.yaml]: Please don't change this

File: web_widget_one2many_product_picker/README.rst [hunk 3]
```diff
@@ -244,6 +247,8 @@ Contributors
     * Pedro M. Baeza
     * David Vidal
 
+* Giovanni Serra <giovanni@gslab.it>
```
Comment (pedrobaeza) [inline web_widget_one2many_product_picker/README.rst:250]: Why this addition? In case of any contribution of this guy, it should be in a separate commit for respecting attribution

File: web_widget_one2many_product_picker/__manifest__.py [hunk 4]
```diff
@@ -6,7 +6,7 @@
     "summary": "Widget to select products on one2many fields",
     "version": "13.0.1.0.0",
     "category": "Website",
-    "author": "Tecnativa, Odoo Community Association (OCA)",
+    "author": "Tecnativa, " "Odoo Community Association (OCA)",
```
Comment (pedrobaeza) [inline web_widget_one2many_product_picker/__manifest__.py]: Don't change this

File: web_widget_one2many_product_picker/i18n/es.po [hunk 5]
```diff
@@ -6,118 +6,143 @@ msgid ""
...
 #. openerp-web
-#: code:addons/web_widget_one2many_product_picker/static/src/js/views/One2ManyProductPicker/record.js:341
+#: code:addons/web_widget_one2many_product_picker/static/src/js/views/One2ManyProductPicker/record.js:0
 #, python-format
 msgid "[No widget %s]"
 msgstr "[Sin widget %s]"
 
-#~ msgid "Add 1"
-#~ msgstr "Añadir 1"
+#~ msgid "HTTP Routing"
```
Comment (pedrobaeza) [inline web_widget_one2many_product_picker/i18n/es.po]: Purge obsolete translations

File: web_widget_one2many_product_picker/models/base.py [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+
+class BaseModel(models.BaseModel):
+    _inherit = "base"
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        """Avoid create lines that have a product currently used when use the product picker"""
+        relation = self.env.context.get("product_picker_relation")
+        if relation == self._name and len(vals_list):
+            product_field = self.env.context.get("product_picker_product_field")
```
Comment (pedrobaeza) [inline web_widget_one2many_product_picker/models/base.py]: Don't repeat code in create and write, but call the same method for each `vals`

**Comment by Tardo** [comment]
Changes done

**Review by pedrobaeza — APPROVED** [review]
